common/persistence/namespaceReplicationQueue.go

```diff
@@ -155,10 +158,9 @@ func (q *namespaceReplicationQueueImpl) GetReplicationMessages(
 		return nil, lastMessageID, err
 	}
 
-	var replicationTasks []*replicationspb.ReplicationTask
+	replicationTasks := make([]*replicationspb.ReplicationTask, 0, len(messages))
```

Comment:
I curious if `make([]*replicationspb.ReplicationTask, len(messages))` and then use indexer instead of `append` in the loop is even faster.

Reply:
```go
func BenchmarkAppend(b *testing.B) {
	count := 1000000
	slice := make([]int, 0, count)

	for i := 0; i < count; i++ {
		slice = append(slice, i)
	}
}

func BenchmarkIndex(b *testing.B) {
	count := 1000000
	slice := make([]int, count)

	for i := 0; i < count; i++ {
		slice[i] = i
	}
}
```

BenchmarkAppend-16    	1000000000	         0.000966 ns/op
BenchmarkIndex-16    	        1000000000	         0.000818 ns/op

no big difference, however the append solution is more resilient to `index out of bound`

Reply:
actually i forgot the for loop for the bench, here is an update
```go
func BenchmarkAppend(b *testing.B) {
	for n := 0; n < b.N; n++ {
		count := 1000
		slice := make([]int, 0, count)

		for i := 0; i < count; i++ {
			slice = append(slice, i)
		}
	}
}

func BenchmarkIndex(b *testing.B) {
	for n := 0; n < b.N; n++ {
		count := 1000
		slice := make([]int, count)

		for i := 0; i < count; i++ {
			slice[i] = i
		}
	}
}
```
BenchmarkAppend-16    	  760594	      1543 ns/op
BenchmarkIndex-16    	  836673	      1387 ns/op